src/scalar_4x64_impl.h

```diff
@@ -812,39 +812,42 @@ static void secp256k1_scalar_mul_512(uint64_t l[8], const secp256k1_scalar *a, c
     /* Extract l7 */
     "movq %%r8, 56(%%rsi)\n"
     : "+d"(pb)
-    : "S"(l), "D"(a->d)
+    : "S"(l8), "D"(a->d)
     : "rax", "rbx", "rcx", "r8", "r9", "r10", "r11", "r12", "r13", "r14", "r15", "cc", "memory");
+
+    SECP256K1_CHECKMEM_MSAN_DEFINE(l8, sizeof(*l8) * 8);
+
 #else
     /* 160 bit accumulator. */
     uint64_t c0 = 0, c1 = 0;
     uint32_t c2 = 0;
 
     /* l[0..7] = a[0..3] * b[0..3]. */
```

Comment:
```suggestion
    /* l8[0..7] = a[0..3] * b[0..3]. */
```
And there are some other comments that may require updating... E.g., `Extract l7`. I admit that these are confusing now that the variable name is `l8`. But I guess it should anyway be `Extract l8[7]`...

Reply:
Ah, I figured comments like `l6` still made sense conceptually, but sure, I'll update them to match.